distributed/worker.py

```diff
@@ -232,6 +234,8 @@ def start_services(self, listen_ip=''):
     def _start(self, addr_or_port=0):
         assert self.status is None
 
+        preload(self.preload, self)
```

Comment:
We might want to call this after we call `listen` just so that the worker has a proper address.  We probably want to call this before `start_services` though.